packages/mdx/index.js

```diff
@@ -103,7 +104,13 @@ function sync(mdx, options) {
   const opts = Object.assign({}, DEFAULT_OPTIONS, options)
   const compiler = createCompiler(opts)
 
-  const { contents } = compiler.processSync(mdx)
+  const fileOpts = { contents: mdx };
+  if (opts.filename) {
+    fileOpts.path = opts.filename;
+  }
+  const file = vfile(fileOpts);
```

Comment:
On thing that may make this cleaner, and remove the `vfile` dependency, is that you pass `fileOpts` directly into `.process` / `.processSync`, no need to wrap yourself!

Reply:
Hmmmmm that's true, that would make the API not backward compatible though because it would need to be wrapped in `{content}`. It feels a bit less straightforward, but I can change it!

Reply:
would it not? `.process` and `.processSync` accept both, and create a vfile from either? Maybe I’m missing something!

Reply:
Whoops, you're right!